### PR TITLE
Impact of Multiple Measured Elements to Conditional Triples

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**
Extending the Compatibility PR to other triples. Once we have a consensus, this PR will be deleted, it will be either merged into
 PR #268

### CODE REVIEW

File: cddl/examples/comid-integrity-registers.diag [hunk 1]
```diff
@@ -20,32 +20,34 @@
...
-        / comid.mval / 1 : {
-          / comid.integrity-registers / 14 : {
-            0: [
-              [
-                / hash-alg-id / 1, / sha256 /
-                / hash-value / h'44aa336af4cb14a879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b'
+      [
+        / measurement-map / {
+          / comid.mval / 1 : {
+            / comid.integrity-registers / 14 : {
```
Comment (deeglaze) [inline cddl/examples/comid-integrity-registers.diag:26]: I'm a little surprised we're keeping integrity-registers when mkey can express the ir name.

Comment (yogeshbdeshpande) [reply cddl/examples/comid-integrity-registers.diag:26]: @deeglaze We can discuss Integrity registers removal in a separate issue and a follow up PR!

I deliberately did not wanted to make things complicated all in one go!

**Comment by nedmsmith** [comment]
Hey, I put together an experiment where I started with the ev and evs cddl and constructed records for conditional endorsed and conditional endorsed series. This CDDL compiles.
```
start = conditional-endorsement-record / conditional-endorsement-series-record

conditional-endorsement-record = [
  conditions: [ + environment-element-map ]
  additions: [ + environment-element-map ]
]

conditional-endorsement-series-record = [
  conditions: [ + environment-element-map ]
  series: [ + conditional-series-map ]
]

environment-element-map = {
  &(environment: 0) => environment-map
  &(elements: 1) => [ + measurement-map ]
}

conditional-series-map = {
  ; first selections that match terminates search
  &(selections: 0) => [ + environment-element-map ]
  &(additions: 1) => [ + environment-element-map ]
}
```
Here are a couple of examples. First for conditional endorsement:
```
/ conditional-endorsement-record / [
    / conditions / [
        / environment-element-map / {
            / environment-map / 0 : {
                / comid.class / 0 : {
                    / comid.class-id / 0 :
                        / tagged-uuid-type / 37(
                        h'67b28b6c34cc40a19117ab5b05911e37'
                        ),
                    / comid.vendor / 1 : "ACME Inc.",
                    / comid.model / 2 : "ACME RoadRunner",
                    / comid.layer / 3 : 1
                }
            },
            / elements / 1 : [
                / measurement-map / {
                    / comid.mval / 1 : {
                        / comid.ver / 0 : {
                            / comid.version / 0 : "1.0.0",
                            / comid.version-scheme / 1 : 16384 / semver /
                        },
                        / comid.digests / 2 : [ [
                            / hash-alg-id / 1, / sha256 /
                            / hash-value / h'44aa336af4cb14a879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b'
                        ] ]
                    }
                }
            ]
        }
    ],
    / additions / [
        / environment-element-map / {
            / environment-map / 0 : {
                / comid.class / 0 : {
                    / comid.class-id / 0 :
                        / tagged-uuid-type / 37(
                        h'67b28b6c34cc40a19117ab5b05911e37'
                        ),
                    / comid.vendor / 1 : "ACME Inc.",
                    / comid.model / 2 : "ACME RoadRunner",
                    / comid.layer / 3 : 1
                }
            },
            / elements / 1 : [
                / measurement-map / {
                    / comid.mval / 1 : {
                        / comid.ver / 0 : {
                            / comid.version / 0 : "1.0.0",
                            / comid.version-scheme / 1 : 16384 / semver /
                        },
                        / comid.digests / 2 : [ [
                            / hash-alg-id / 1, / sha256 /
                            / hash-value / h'44aa336af4cb14a879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b'
                        ] ]
                    }
                }
            ]
        }
    ]
]
```
This is for conditional endorsement series:
```
/ conditional-endorsement-series-record / [
    / conditions / [
        / environment-element-map / {
            / environment-map / 0 : {
                / comid.class / 0 : {
                    / comid.class-id / 0 :
                        / tagged-uuid-type / 37(
                        h'67b28b6c34cc40a19117ab5b05911e37'
                        ),
                    / comid.vendor / 1 : "ACME Inc.",
                    / comid.model / 2 : "ACME RoadRunner",
                    / comid.layer / 3 : 1
                }
            },
            / elements / 1 : [
                / measurement-map / {
                    / comid.mval / 1 : {
                        / comid.ver / 0 : {
                            / comid.version / 0 : "1.0.0",
                            / comid.version-scheme / 1 : 16384 / semver /
                        },
                        / comid.digests / 2 : [ [
                            / hash-alg-id / 1, / sha256 /
                            / hash-value / h'44aa336af4cb14a879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b'
                        ] ]
                    }
                }
            ]
        }
    ],
    / series / [
        / conditional-series-map / {
            / selections / 0 : [
                / environment-element-map / {
                    / environment-map / 0 : {
                        / comid.class / 0 : {
                            / comid.class-id / 0 :
                                / tagged-uuid-type / 37(
                                h'67b28b6c34cc40a19117ab5b05911e37'
                                ),
                            / comid.vendor / 1 : "ACME Inc.",
                            / comid.model / 2 : "ACME RoadRunner",
                            / comid.layer / 3 : 1
                        }
                    },
                    / elements / 1 : [
                        / measurement-map / {
                            / comid.mval / 1 : {
                                / comid.ver / 0 : {
                                    / comid.version / 0 : "1.0.0",
                                    / comid.version-scheme / 1 : 16384 / semver /
                                },
                                / comid.digests / 2 : [ [
                                    / hash-alg-id / 1, / sha256 /
                                    / hash-value / h'44aa336af4cb14a879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b'
                                ] ]
                            }
                        }
                    ]
                }
            ],
            / additions / 1 : [
                / environment-element-map / {
                    / environment-map / 0 : {
                        / comid.class / 0 : {
                            / comid.class-id / 0 :
                                / tagged-uuid-type / 37(
                                h'67b28b6c34cc40a19117ab5b05911e37'
                                ),
                            / comid.vendor / 1 : "ACME Inc.",
                            / comid.model / 2 : "ACME RoadRunner",
                            / comid.layer / 3 : 1
                        }
                    },
                    / elements / 1 : [
                        / measurement-map / {
                            / comid.mval / 1 : {
                                / comid.ver / 0 : {
                                    / comid.version / 0 : "1.0.0",
                                    / comid.version-scheme / 1 : 16384 / semver /
                                },
                                / comid.digests / 2 : [ [
                                    / hash-alg-id / 1, / sha256 /
                                    / hash-value / h'44aa336af4cb14a879432e53dd6571c7fa9bccafb75f488259262d6ea3a4d91b'
                                ] ]
                            }
                        }
                    ]
                }
            ]
        }
    ]
]
```

**Comment by yogeshbdeshpande** [comment]
@nedmsmith Thanks. The point is : It is clear that the multiplicity of measurement-map, is required on the stateful-environment, as well as `conditionally-endorsed triples`. However,  I am not clear, how the multiplicity is relevant and justified for the conditional endorsement-series triple. Can you please add an explanation to issue #288. We can align the CDDL then...

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1217,14 +1217,18 @@ matched.
 ~~~
 
 The series object is an array of `conditional-series-record` that has both Reference and Endorsed Values.
-Each `conditional-series-record` record is evaluated in the order it appears in the series array.
+
+Each `conditional-series-record` has measurements belonging to one or more measured elements, pertaining to the environment that appears in the stateful-environment record.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1221]:
```suggestion
For each `conditional-series-record` entry, if the `selection` matches the chosen ACS entry, the `addition` is added to the ACS.
```

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
+
 The Endorsed Values are accepted if the series condition in a `conditional-series-record` matches the ACS.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1223]:
```suggestion

```

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
-The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted.
+The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted. In order for a `conditional-series-record` to match, measurements pertaining to every measured element in the record MUST match.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1224]:
```suggestion
The first `conditional-series-record` entry that successfully matches the chosen ACS entry terminates the series. 
For a `conditional-series-record` to match, every measurement in the `measurement-map` list MUST match a measurement in the chosen ACS entry.
```

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
+
 If none of the series conditions match an ACS Entry, the triple is not matched,
 and no Endorsed values are accepted.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1227]:
```suggestion
and no `addition` values are accepted.
```

File: draft-ietf-rats-corim.md [hunk 6]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
 The series object is an array of `conditional-series-record` that has both Reference and Endorsed Values.
-Each `conditional-series-record` record is evaluated in the order it appears in the series array.
+
+Each `conditional-series-record` has measurements belonging to one or more measured elements, pertaining to the environment that appears in the stateful-environment record.
+
 The Endorsed Values are accepted if the series condition in a `conditional-series-record` matches the ACS.
-The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted.
+The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted. In order for a `conditional-series-record` to match, measurements pertaining to every measured element in the record MUST match.
+
 If none of the series conditions match an ACS Entry, the triple is not matched,
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1226]:
```suggestion
If none of the `selection` values match in the chosen ACS entry, the triple is not matched,
```

File: draft-ietf-rats-corim.md [hunk 7]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
 and no Endorsed values are accepted.
 
 The `authorized-by` value in `measurement-map` in the stateful environment, if present,
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1229]:
```suggestion
If the `authorized-by` value is present in the triple `condition` (i.e., in the `measurement-map` of the `stateful-environment-record`),
all authority entries MUST be present in the ACS entry, otherwise the ACS entry does not match.
```

File: draft-ietf-rats-corim.md [hunk 8]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
-applies to all measurements in the triple, including `conditional-series-record` records.
+applies to reference value measurements in the triple, for `conditional-series-record` records.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1230]:
```suggestion
If the series `selection` populates `authorized-by`, the chosen ACS MUST contain the same measurements and authority as contained in the `selection` entry.
```

File: draft-ietf-rats-corim.md [hunk 9]
```diff
@@ -1217,14 +1217,18 @@ matched.
...
+If the `authorized-by` value is provided in the `measurement-map` for endorsed values, they get added to the ACS.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1231]:
```suggestion
If the series `addition` entry contains `authorized-by` values, they are ignored.
```

Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1231]: Note: authority semantics are still being discussed. It is safer for this triple to ignore them in the addition until the semantics for series authority is determined.

**Review by nedmsmith — COMMENTED** [review]
Several suggested changes.

File: cddl/conditional-series-record.cddl [hunk 10]
```diff
@@ -1,6 +1,6 @@
 conditional-series-record = [
   ; reference values to be matched against evidence
```
Comment (nedmsmith) [inline cddl/conditional-series-record.cddl:2]:
```suggestion

```

Comment (nedmsmith) [reply cddl/conditional-series-record.cddl:2]: The prose of the document describes the semantics

**Review by nedmsmith — COMMENTED** [review]
multiple

File: draft-ietf-rats-corim.md (hunk 8 above)
Comment (deeglaze) [reply draft-ietf-rats-corim.md:1230]: I think we want to say matching authority so we can punt on the matching semantics. Sounds like you want subset ([⊆](https://www.compart.com/en/unicode/U+2286)) and Andy wants nonempty intersection (often notated with [⋈](https://www.compart.com/en/unicode/U+22C8)). This is a good place to use tagging for pattern operators so both forms can be expressible.

File: draft-ietf-rats-corim.md (hunk 9 above)
Comment (deeglaze) [reply draft-ietf-rats-corim.md:1231]: I think since it's a non-extensible type, you can use a type like element-map (but a copy to not confuse internal and external representations) to drop the authorities. You can add them back with a spec update.

File: draft-ietf-rats-corim.md (hunk 6 above)
Comment (deeglaze) [reply draft-ietf-rats-corim.md:1226]: Are we expressing the semantics in terms of the triple or in terms of the relation of internal representations we translate triples to? I think we should be consistent and use the internal representation for defining semantics.

File: cddl/conditional-series-record.cddl [hunk 11]
```diff
@@ -1,6 +1,6 @@
...
-  refv: measurement-values-map
+  refv: [ +measurement-map]
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply cddl/conditional-series-record.cddl:3]: nit: the spacing for other one-or-more lists has a space on both sides of the + as well.

File: cddl/conditional-series-record.cddl [hunk 12]
```diff
@@ -1,6 +1,6 @@
...
   ; endorsed values that apply in case revf matches
-  endv: measurement-values-map
+  endv: [ +measurement-map]
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply cddl/conditional-series-record.cddl:5]: nit: same as above " + "

**Review by nedmsmith — COMMENTED** [review]
See suggested changes

**Comment by yogeshbdeshpande** [comment]
@deeglaze & @nedmsmith Thanks for the comments, I will review and work on them!

File: draft-ietf-rats-corim.md (hunk 9 above)
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1231]: ok, for now I will accept this suggestion , but really feel this is required. We will discuss the same in a separate PR

File: draft-ietf-rats-corim.md (hunk 3 above)
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1224]: Why have you deleted this?

Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1224]: ok, understood the concept that in series, the first condition match terminates the series, to line 1224 is not really applicable!

**Review by nedmsmith — APPROVED** [review]

**Comment by yogeshbdeshpande** [comment]
@henkbirkholz @thomas-fossati & @andrew-draper : kindly review this.

@deeglaze Thanks for your review. I have incorporated all the comments which you had. Kindly check once and let me know, if anything is outstanding!

**Comment by yogeshbdeshpande** [comment]
> LGTM, thanks.
> 
> I am unsure about the use of the qualifier "chosen" when talking about the ACS entry that is being compared against. There is no selection criterion apart from picking one after the other from the set, no?
> 
> Inlined a few tiny editorial change requests.

I think, you are correct, the term `chosen` is a bit mis-leading.

The matching condition from this Triple to Evidence entry is the only criterion to get access to the correct entry in the ACS.
I will remove the word chosen and still it would make sense I suppose!

File: draft-ietf-rats-corim.md [hunk 13]
```diff
@@ -1216,15 +1216,21 @@ matched.
...
 ~~~
 
-The series object is an array of `conditional-series-record` that has both Reference and Endorsed Values.
-Each `conditional-series-record` record is evaluated in the order it appears in the series array.
-The Endorsed Values are accepted if the series condition in a `conditional-series-record` matches the ACS.
-The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted.
-If none of the series conditions match an ACS Entry, the triple is not matched,
-and no Endorsed values are accepted.
+The series is an array of `conditional-series-record` that has a `selection`, and an `addition`, both expressed as a list of `measurement-map`.
+The `selection` and `addition` operate within the scope of the conditional endorsement series triple record's matching `condition`.
```
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md]: I have made it more explicit and crisp, which is I suppose you wanted. Please check!

**Comment by nedmsmith** [comment]
With these changes, `stateful-environment-map` is identical to `endorsed-triple-record`. Should they be merged?

**Comment by yogeshbdeshpande** [comment]
> With these changes, `stateful-environment-map` is identical to `endorsed-triple-record`. Should they be merged?

Not really, the construct looks identical, but internally they have a very different meaning and hence interpretation.

Its just so happens that we have an identical map for `reference-values` and `endorsed-values`, which creates this ambiguity.

But I would keep them separate.

**Comment by nedmsmith** [comment]
> Its just so happens that we have an identical map for reference-values and endorsed-values

Actually, there's no such thing as `reference-values` or `endorsed-values`. We have triples for `reference-triple-record` and `endorsed-triple-record`, but these are the structures that define the triple (i.e., subject, object, predicate). Since the predicate is implied, the name of the structure serves as a pneumonic for the triple's predicate. 

The statement `endorsed-triple-record` is being repurposed from a stand-alone triple to something that has different meaning when inside of a `conditional-endorsed-triple-record`. Specifically, the condition in a `endorsed-triple-record` triple is not applied when it appears as the addition part of a `conditional-endorsed-triple-record`.  (i.e., it isn't actually a triple inside of a triple).

The prose (and CBOR encoding) that describes:
```
conditional-endorsement-triple-record = [
  conditions: [ + stateful-environment-record ]
  additions: [ + endorsed-triple-record ]
]
```
will be identical to the prose (CBOR) that describes the following (sans cddl names):
```
conditional-endorsement-triple-record = [
  conditions: [ + stateful-environment-record ]
  additions: [ + stateful-environment-record ]
]
```
Since `stateful-environment-record` doesn't double as a triple, it has no implied semantics on its own. It is only meaningful in the context of a triple description. 

Interestingly, it will be identical to the prose (CBOR) that describes this statement too (sans cddl names):
```
conditional-endorsement-triple-record = [
  conditions: [ + endorsed-triple-record ]
  additions: [ + endorsed-triple-record ]
]
```
However, the reader is assumed to understand that we don't actually mean apply the semantics of the endorsed-triple-record triple, but apply the semantics that the prose for the conditional-endorsed-triple defines.

My point is we're mixing and matching reader expectations given the new definition of stateful-environment-map but offer no intuition as to why. With the addition of the label, the semantics are more easily communicated via the label, rather than the structure name and since stateful-environment-map is just building block CDDL, the second option seems to be the most intuitive of the three.

**Comment by thomas-fossati** [comment]
> With these changes, `stateful-environment-map` is identical to `endorsed-triple-record`. Should they be merged?

Interesting observation.  Isn't this a manifestation of special casing `endorsed-triple-record` instead of treating it as a conditional endorsement?

File: cddl/conditional-series-record.cddl [hunk 14]
```diff
@@ -1,6 +1,5 @@
 conditional-series-record = [
-  ; reference values to be matched against evidence
-  refv: measurement-values-map
+  selection: [ + measurement-map]
```
Comment (andrew-draper) [inline cddl/conditional-series-record.cddl:2]: Isn't this just more conditions?
Would it be better to name it condition?
```suggestion
  condition: [ + measurement-map]
```

File: draft-ietf-rats-corim.md [hunk 15]
```diff
@@ -1216,15 +1216,19 @@ matched.
...
 ~~~
 
-The series object is an array of `conditional-series-record` that has both Reference and Endorsed Values.
-Each `conditional-series-record` record is evaluated in the order it appears in the series array.
-The Endorsed Values are accepted if the series condition in a `conditional-series-record` matches the ACS.
-The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted.
-If none of the series conditions match an ACS Entry, the triple is not matched,
-and no Endorsed values are accepted.
+The series is an array of `conditional-series-record` that has a `selection` and an `addition`, both expressed as a list of `measurement-map`.
+The `selection` and `addition` operate within the scope of the `conditional-endorsement-series-triple-record`'s matching `condition`.
```
Comment (andrew-draper) [inline draft-ietf-rats-corim.md:1220]:
```suggestion
The `selection` and `addition` operate within the scope of the environment-map found within the `conditional-endorsement-series-triple-record`'s `condition`.
```

File: draft-ietf-rats-corim.md [hunk 16]
```diff
@@ -1216,15 +1216,19 @@ matched.
...
-The `authorized-by` value in `measurement-map` in the stateful environment, if present,
-applies to all measurements in the triple, including `conditional-series-record` records.
+For each `conditional-series-record` entry, if the `selection` matches the ACS entry, the `addition` is added to the ACS.
+
+The first `conditional-series-record` entry that successfully matches the ACS entry terminates the series.
+For a `conditional-series-record` to match, every measurement in the `measurement-map` list MUST match a measurement in the ACS entry.
+
+If none of the `selection` values match in the ACS entry, the triple is not matched, and no `addition` values are accepted.
+
+If the `authorized-by` value is present in the triple `condition` (i.e., in the `measurement-map` of the `stateful-environment-record`), all authority entries of the `condition` MUST be present in the ACS entry, otherwise the ACS entry does not match.
```
Comment (andrew-draper) [inline draft-ietf-rats-corim.md:1229]: Is this needed here>
Should we be discussing how to transform external encoding into ECT encoding so that the text in 8.9 (which already says this) can be used?

**Review by andrew-draper — APPROVED** [review]
As far as the CDDL is concerned, I think this would be ok.
I think more work is needed on the test describing the conditions, it duplicates (and potentially disagrees with) the ACS comparison text later in the document.

File: draft-ietf-rats-corim.md (hunk 16 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1229]: > Is this needed here> Should we be discussing how to transform external encoding into ECT encoding so that the text in 8.9 (which already says this) can be used?

I think it's OK to describe the semantics of the triple in terms of what to expect will be reflected in the ACS. The transformation section describes how to map from external to internal. The Phase 4 processing section describes how the ACS realizes an ending state. This section describes what to expect in terms of the ending state in ACS - i.e., it's a what vs. how difference.

**Comment by nedmsmith** [comment]
> > With these changes, `stateful-environment-map` is identical to `endorsed-triple-record`. Should they be merged?
> 
> Interesting observation. Isn't this a manifestation of special casing `endorsed-triple-record` instead of treating it as a conditional endorsement? Or is it a genuine reification of the `endorsed-triple-record`?

It is special casing `endorsed-triple-record` because it doesn't treat the `environment-map` in `endorsed-triple-record` as a `condition` only as an `addition`.  The processing described in Endorsement Triples Transformations {#sec-end-trans} isn't applied to `endorsed-triple-record` when it appears in a `conditional-endorsed-triple-record`.

**Review by nedmsmith — COMMENTED** [review]
Fixed typo in transformations section

File: draft-ietf-rats-corim.md [hunk 17]
```diff
@@ -1216,15 +1216,21 @@ matched.
...
-The Endorsed Values are accepted if the series condition in a `conditional-series-record` matches the ACS.
-The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted.
-If none of the series conditions match an ACS Entry, the triple is not matched,
-and no Endorsed values are accepted.
+The series is an array of `conditional-series-record` that has a `selection`, and an `addition`, both expressed as a list of `measurement-map`.
+The `selection` and `addition` operate within the scope of the conditional endorsement series triple record's matching `condition`.
 
-The `authorized-by` value in `measurement-map` in the stateful environment, if present,
-applies to all measurements in the triple, including `conditional-series-record` records.
+For each `conditional-series-record` entry, if the `selection` matches the chosen ACS entry, the `addition` is added to the ACS.
```
[the remark this answers is not in the document]
Comment (nedmsmith) [reply draft-ietf-rats-corim.md]: > Addressed: Remove the chosen term!

The original text was trying to capture the detail that the selection scope is within the range of the `condition` and that the `addition` is within the same range.  This is a necessity since the series records don't have a qualifying environment-map.

**Review by henkbirkholz — APPROVED** [review]
